packages/flare/python/cron_job_ingest_events.py

```diff
@@ -1,72 +1,207 @@
+    ):
+        pass
 
-    if event_feed["items"]:
-        for item in event_feed["items"]:
-            print(json.dumps(item))
+    api_key: Optional[str] = None
+    tenant_id: Optional[int] = None
+    for item in app.service.storage_passwords.list():
+        if item.content.username == PasswordKeys.API_KEY.value:
+            api_key = item.clear_password
 
+        if item.content.username == PasswordKeys.TENANT_ID.value:
+            tenant_id = (
+                int(item.clear_password) if item.clear_password is not None else None
+            )
+
+    if not api_key:
+        raise Exception("API key not found")
+
+    if not tenant_id:
+        raise Exception("Tenant ID not found")
+
+    try:
+        flare_api = FlareAPI(app=app, api_key=api_key, tenant_id=tenant_id)
+
+        next = get_next(app=app, tenant_id=tenant_id)
+        start_date = get_start_date(app=app)
+        for response in flare_api.retrieve_feed(next=next, start_date=start_date):
+            save_last_fetched(app=app)
+
+            # Rate limiting.
+            time.sleep(1)
+
+            if response.status_code != 200:
+                # Use logger here
+                print(response.text, file=sys.stderr)
+                return
+
+            event_feed = response.json()
+            save_start_date(app=app, tenant_id=tenant_id)
+            save_next(app=app, tenant_id=tenant_id, next=event_feed["next"])
+
+            if event_feed["items"]:
+                for item in event_feed["items"]:
+                    print(json.dumps(item))
+    except Exception as e:
+        # Adding logger in next PR
+        print("Exception={}".format(e))
+
+
+def get_next(app: client.Application, tenant_id: int) -> Optional[str]:
+    return get_collection_value(
+        app=app, key="{}{}".format(CollectionKeys.NEXT_TOKEN.value, tenant_id)
+    )
 
-def get_from_value(app: client.Application) -> Optional[str]:
-    from_: Optional[str] = None
-    if KV_COLLECTION_NAME in app.service.kvstore:
-        data = app.service.kvstore[KV_COLLECTION_NAME].data.query()
-        if len(data) > 1:
-            from_ = data[0]["value"]
-        else:
-            app.service.kvstore.delete(KV_COLLECTION_NAME)
 
-    return from_
+def get_start_date(app: client.Application) -> Optional[str]:
+    return get_collection_value(app=app, key=CollectionKeys.START_DATE.value)
 
 
-def set_from_value(app: client.Application, next: Optional[str]):
+def get_current_tenant_id(app: client.Application) -> Optional[int]:
+    current_tenant_id = get_collection_value(
+        app=app, key=CollectionKeys.CURRENT_TENANT_ID.value
+    )
+    return int(current_tenant_id) if current_tenant_id else None
+
+
+def get_last_fetched(app: client.Application) -> Optional[int]:
+    timestamp_last_fetched = get_collection_value(
+        app=app, key=CollectionKeys.TIMESTAMP_LAST_FETCH.value
+    )
+    return int(timestamp_last_fetched) if timestamp_last_fetched else None
```

Comment:
We need to handle bad data here. Let's maybe wrap this in a try-except just to make sure the app still works if there is a bad value in get_collection_value